extensions/opentelemetry/opentelemetry-async-mutiny/runtime/src/main/java/io/quarkus/opentelemetry/async/mutiny/runtime/tracing/OpenTelemetryMultiInterceptor.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.opentelemetry.async.mutiny.runtime.tracing;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+import io.opentelemetry.context.Context;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.infrastructure.MultiInterceptor;
+
+public class OpenTelemetryMultiInterceptor implements MultiInterceptor {
+
+    @Override
+    public <T> Multi<T> onMultiCreation(final Multi<T> multi) {
+        if (multi instanceof TracingMulti) {
+            return multi;
+        }
+        return new TracingMulti<>(Context.current(), multi);
```

Comment:
Are you sure the Context.current is what you want? 

It will be the current at assembly time. I'm pretty sure you need the one at subscription time.

Reply:
Hmm. Long time ago... I cannot really remember. It is also present in onSubscription. I mean the tests are there, so feel free to improve the approach. I quickly checked the rxjava impl. I do not see it there. There the current context is only used at subscription time. I think something was broken when I did not do this, but as said, I cannot really remember. I had some difficulties with the Multi instrumentation, that is at least what I can remember.